### DISCUSSION TITLE
How to use "extendRoutes"?

### DISCUSSION THREAD

**Discussion by crotanite** [nuxt/nuxt#16397] [Questions]
The documentation for modules says there is an "extendRoutes" function, but this doesn't seem to be importable at all through "@nuxt/kit".

Is it currently possible to use this?

**Comment by wobsoriano** [comment]
It's not importable through `@nuxt/kit`, but you mutate it directly via second param of setup method when defining a module:

```ts
import {
    defineNuxtModule
} from '@nuxt/kit'

export default defineNuxtModule({
  name: 'your-module',
  setup(options, nuxt) {
    nuxt.options.router.extendRoutes = (routes, resolve) => {
        routes.push({
            name: 'custom',
            path: '*',
            component: resolve(__dirname, 'pages/404.vue')
        })
    }
  }
});
```

**Reply by crotanite** [reply]
This doesn't seem to do anything. I reproduced it here:
https://codesandbox.io/s/xenodochial-kate-n5k7x?file=/modules/module.ts 

It never hits the `console.log('here')` and the route isn't added to the routes list.

**Reply by wobsoriano** [reply]
Idk why it's not logging in codesandbox, but you should get logs in the console in local development. That's how you access the `extendRoutes` property tho, same with Nuxt 2

**Reply by crotanite** [reply]
I've just been trying it in my own build, and its not working there either.

**Reply by alterhu2020** [reply]
Hey Team, any way we can extend the `vue-router` to add addtional custom router. thanks for help
